server/src/test/java/org/elasticsearch/action/admin/indices/settings/put/UpdateSettingsRequestTests.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.settings.put;
+
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractXContentTestCase;
+
+import java.io.IOException;
+import java.util.function.Predicate;
+
+public class UpdateSettingsRequestTests extends AbstractXContentTestCase<UpdateSettingsRequest> {
+
+    private boolean enclosedSettings = randomBoolean();
+
+    @Override
+    protected UpdateSettingsRequest createTestInstance() {
+        UpdateSettingsRequest testRequest = UpdateSettingsRequestStreamableTests.createTestItem();
+        if (enclosedSettings) {
+            UpdateSettingsRequest requestWithEnclosingSettings = new UpdateSettingsRequest(testRequest.settings()) {
+                public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+                    builder.startObject();
+                    builder.startObject("settings");
+                    this.settings().toXContent(builder, params);
+                    builder.endObject();
+                    builder.endObject();
+                    return builder;
+                }
+            };
+            return requestWithEnclosingSettings;
+        }
+        return testRequest;
+    }
+
+    @Override
+    protected UpdateSettingsRequest doParseInstance(XContentParser parser) throws IOException {
+        return new UpdateSettingsRequest().fromXContent(parser);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return enclosedSettings;
```

Comment:
I am fine either way. I decided to leave it here till #29268 is fixed and the enclosing `"settings"` are removed, then the test can be really simplified :)

IMHO for any request this should be set to `false` : any object insertion at the top level should make the parsing fail ( for predefined field names ). 

Off-topic : should we consider extending the `AbstractXContentTestCase` to explicitly test that if `supportsUnknownFields` is `false` actually an exception is thrown when an object is inserted at the top-level ? ( this can lead to a lot of changes in the tests and maybe the code, but could help determine how strict the rest request parsing is... )

Reply:
I would set it to false too. I think that what you propose is a good idea, we have been using that flag only to test that we are lenient with responses, but we could use it also to test that we are strict.

Reply:
what I had missed is that we cannot set it to false unless we get rid of the leniency caused by parsing the settings object etc.

Reply:
Do you want to open an issue about this ?

Reply:
yes makes sense to me.